be/src/vec/exec/join/process_hash_table_probe_impl.h

```diff
@@ -177,22 +177,51 @@ Status ProcessHashTableProbe<JoinOpType>::do_process(HashTableType& hash_table_c
     KeyGetter key_getter(probe_raw_ptrs, _join_node->_probe_key_sz, nullptr);
 
     if (probe_index == 0) {
-        size_t old_probe_keys_memory_usage = 0;
-        if (_arena) {
-            old_probe_keys_memory_usage = _arena->size();
+        if (!_arena) {
+            _arena.reset(new Arena());
```

Comment:
maybe we could not arena memory reuse? To avoid allocate memory too frequently and many page fault.